### ISSUE TITLE
How can I compute a meaningful clip_norm threshold for my particular Network?

### ISSUE THREAD

**Issue by erogol** [fchollet/keras#28]

**Comment by fchollet** [comment]
Absent additional information, this is difficult to answer. Norm clipping is typically useful for RNNs. Is this what you are working with?

**Comment by erogol** [comment]
4 layers feedforward network. I tried all I know with keras but I cannot improve the results, my network still below Gradient Boosted Trees but It should be comparable at least with all the theory behind deep learning. I try to regularize my network with clipping. Maybe keras might provide a method to check the amount of gradient per iteration for each unit like caffe.

**Comment by fchollet** [comment]
If you perform poorly on the training data, you need a bigger/deeper net. If you perform well on the training data, but poorly on the test data, you need more overfitting prevention (Dropout, L1 reg, L2 reg) or/and a smaller net.

You might also want to optimize your network architecture and hyperparameters using something like Spearmint. https://github.com/JasperSnoek/spearmint

Note that not all problems are solvable with neural nets (if the amount of training data is fixed); the applicability of neural nets will often depend on the quantity/quality of training data available.

**Comment by erogol** [comment]
I see thanks for your comments, in any way I try to squeeze as much as possible. However, as you said I guess the problem is not suitable for NN since larger net even gives lower results compared to one hidden layer net.

**Comment by 3rd3** [comment]
Plot some gradient histograms of the concatenated values all variables if you want to set this threshold for all gradients at once; or for each variable separately. That will give you a good idea how to set the clipping parameter.
